i18n/detect.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+"""
+Function to check whether or not the English strings are up to date
+"""
+from __future__ import print_function
+from subprocess import CalledProcessError
+
+from i18n import Runner
+from i18n.execute import execute
+from i18n.extract import Extract
+
+
+class Detect(Runner):
+    """
+    Class used to check if the English source strings are up to date
+    """
+    def run(self, args):
+        """
+        Main entry point of script
+        """
+        Extract().run(args)  # Run extract command
+        changes = self.detect_changes()
+        message = self.get_message(changes)
+        print(message)
+
+    def detect_changes(self):
+        """
+        Detect if extract command changed msgid or msgstr strings.
```

Comment:
```python
Detect if changes have been made to the msgid or msgstr lines in the translation files.

Returns:
  integer: 1, if changes detected; otherwise, 0.

Note:
  This method requires ``git`` to be installed on the executing machine.
```